tests/test_model_name.py

```diff
@@ -1,7 +1,74 @@
-# TODO: Unit test get_model_name
+import warnings
+
+import factory
+import pytest
+
+from pytest_factoryboy.fixture import get_model_name, named_model
 from tests.compat import assert_outcomes
 
 
+def make_class(name: str):
+    """Create a class with the given name."""
+    return type(name, (object,), {})
+
+
+@pytest.mark.parametrize("model_cls", [dict, set, list, frozenset, tuple])
+def test_get_model_name_warns_for_common_containers(model_cls):
+    """Test that a warning is raised when common containers are used as models."""
+
+    class ModelFactory(factory.Factory):
+        class Meta:
+            model = model_cls
+
+    with pytest.warns(
+        UserWarning,
+        match=rf"Using a .*{model_cls.__name__}.* as model type for .*ModelFactory.* is discouraged",
+    ):
+        assert get_model_name(ModelFactory)
+
+
+def test_get_model_name_does_not_warn_for_user_defined_models():
+    """Test that no warning is raised for when using user-defined models"""
+
+    class Foo:
+        pass
+
+    class ModelFactory(factory.Factory):
+        class Meta:
+            model = Foo
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        assert get_model_name(ModelFactory) == "foo"
```

Comment:
NIP: to make this test more explicit and obvious we can check emitted warnings in an assert statement explicitly:

```python
    with pytest.warns(None) as recorded_warnings:
        assert get_model_name(ModelFactory) == "foo"
        assert not recorded_warnings
```

We can also pass the exact warning class (`UserWarning` in our case) instead of `None`.

Reply:
this is the way that pytest suggests: https://docs.pytest.org/en/stable/how-to/capture-warnings.html#:~:text=To%20ensure%20that%20no%20warnings%20are%20emitted%2C%20use%3A

Reply:
Oh nice, somehow I missed it in their docs, thanks!